internal/datasets/subjectsetbytype.go

```diff
@@ -19,24 +19,24 @@ func NewSubjectByTypeSet() *SubjectByTypeSet {
 }
 
 // AddSubjectOf adds the subject found in the given relationship, along with its caveat.
-func (s *SubjectByTypeSet) AddSubjectOf(relationship *core.RelationTuple) error {
-	return s.AddSubject(relationship.Subject, relationship.Caveat)
+func (s *SubjectByTypeSet) AddSubjectOf(relationship tuple.Relationship) error {
+	return s.AddSubject(relationship.Subject, relationship.OptionalCaveat)
 }
 
 // AddConcreteSubject adds a non-caveated subject to the set.
-func (s *SubjectByTypeSet) AddConcreteSubject(subject *core.ObjectAndRelation) error {
+func (s *SubjectByTypeSet) AddConcreteSubject(subject tuple.ObjectAndRelation) error {
 	return s.AddSubject(subject, nil)
 }
 
 // AddSubject adds the specified subject to the set.
-func (s *SubjectByTypeSet) AddSubject(subject *core.ObjectAndRelation, caveat *core.ContextualizedCaveat) error {
-	key := tuple.JoinRelRef(subject.Namespace, subject.Relation)
+func (s *SubjectByTypeSet) AddSubject(subject tuple.ObjectAndRelation, caveat *core.ContextualizedCaveat) error {
```

Comment:
Only reason to have a reference receiver is if its needed to mutate the state of the object *or* copying the object will be prohibitively expensive; since the struct here is relatively small, changing to it being a non-reference means that it should be placed on the stack vs the heap